restcomm/restcomm.mscontrol.mms/src/main/java/org/restcomm/connect/mscontrol/mms/MgcpMediaGroup.java

```diff
@@ -220,29 +232,28 @@ protected void play(final Object message) {
         final List<URI> uris = request.uris();
         final int iterations = request.iterations();
         final org.restcomm.connect.mgcp.Play play = new org.restcomm.connect.mgcp.Play(uris, iterations);
-        this.lastEvent = AUMgcpEvent.aupa;
         stop(lastEvent);
+        this.lastEvent = AUMgcpEvent.aupa;
         this.originator = sender();
         ivr.tell(play, self);
         ivrInUse = true;
     }
 
     @SuppressWarnings("unchecked")
     protected void notification(final Object message) {
-        final IvrEndpointResponse<String> response = (IvrEndpointResponse<String>) message;
+        final IvrEndpointResponse response = (IvrEndpointResponse) message;
         final ActorRef self = self();
-        MediaGroupResponse<String> event = null;
+        MediaGroupResponse<CollectedResult> event;
         if (response.succeeded()) {
-            event = new MediaGroupResponse<String>(response.get());
+            event = new MediaGroupResponse<>(response.get());
         } else {
-            event = new MediaGroupResponse<String>(response.cause(), response.error());
+            event = new MediaGroupResponse<>(response.cause(), response.error());
         }
-        // for (final ActorRef observer : observers) {
-        // observer.tell(event, self);
-        // }
         if (originator != null)
             this.originator.tell(event, self);
-        ivrInUse = false;
+        if (!response.get().isPartial()) {
```

Comment:
Changed to `if (response.get() == null || !response.get().isPartial())`
https://github.com/RestComm/Restcomm-Connect/pull/2390/commits/b2d39409adf99b27d3b2d95dd3ea19b90283b953